space-concordia-robotics/robotics-prototype
# Save correct rotation 315

# Assignee Section

## Description
Added correct rotation according to the one shown at the GUI to screenshots and recordings of streams. Passed the rotation value from the javascript to the python. Used ffmpeg filters transpose and vflip to rotate the streams. They work altering a stream already created.

- Added ffmpeg-python library
- Divided filename variable in stream_capture.py in two different variables to avoid confusion with image and video filename
- Created rotation function using transpose and vflip filters
- Made the video filename global to use it in the rotation function as the number of seconds had to equal

Comment: Need to `pip install -r requirements.txt` on this branch before testing for this feature to work.

### Steps for Testing
Run

1. `rosgui`
2. `startgui`
3. `roslaunch robot/util/rosRoverStart/local_rover.launch`
4. Visit `localhost:5000` and follow the steps below

#### Test cases with 1 camera

This can be tested on any page with at least one camera panel.

The location of the saved images/videos will be logged in the GUI console.

1. Turn on stream, rotate image, take a screenshot, verify saved image is correctly rotated.
2. Turn on stream, rotate image, start recording a video, stop recording, verify saved video is correctly rotated.
3. Repeat step 1. and 2. for all 4 orientations.

#### Test cases with 2 cameras

This needs to be tested on the multiple streams page.

1. Open 2 camera feeds: Start recording feed1 first, then start recording feed2
2. Stop recording feed2 first, then stop recording feed1
3. Verify saved videos are saved correctly rotated, repeat for different orientations

Note that the video saved will be saved only in the initial orientation, so flipping the orientation during recording will not affect the saved video.

closes #315

The approval from all software team leads is necessary before merging.

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.


For Pull Requests that do not include code changes, it is not required to perform the tests above.
